src/tcache.c

```diff
@@ -710,16 +807,24 @@ tsd_tcache_data_init(tsd_t *tsd) {
 	tcache_t *tcache = tsd_tcachep_get_unsafe(tsd);
 
 	assert(cache_bin_still_zero_initialized(&tcache->bins[0]));
-	size_t alignment = tcache_bin_alloc_alignment;
-	size_t size = sz_sa2u(tcache_bin_alloc_size, alignment);
+	assert(tcache != NULL);
```

Comment:
nit: not really useful. It's accessed in the assert above already, and it's simply an offset based on tsd (which is not tsdn).